lib/internal/v8.js

```diff
@@ -1,21 +1,54 @@
 'use strict';
 
-function take(it, n) {
-  const result = [];
-  for (const e of it) {
-    if (--n < 0)
-      break;
-    result.push(e);
-  }
-  return result;
+/**
+ * @description Get a clone of the passed in map iterator.
+ *
+ * @param {MapIterator} it The iterator that should be cloned.
+ * @returns {MapIterator}
+ */
```

Comment:
I don't believe we have used JSDoc annotations anywhere else in the code base. Maybe just use prose instead? (I.e., replace this comment block with just `Clone the provided Map Iterator`.)

Speaking of documentation, I think a general comment for this entire file would be good. Specifically:

1. what it is, where these functions are used, maybe a half-sentence intro on V8's natives syntax, and
2. how it is special compared to rest of the code base (loaded before anything else with a special flag that allows natives syntax).